tests/test_clients.py

```diff
@@ -0,0 +1,52 @@
+import time
+
+import jwt
+import pytest
+import requests_mock
+from freezegun import freeze_time
+
+from zgw_consumers.client import build_client
+from zgw_consumers.constants import AuthTypes
+from zgw_consumers.nlx import NLXWithRefreshClient
+from zgw_consumers.test.factories import ServiceFactory
+
+
+@pytest.mark.django_db
+def test_retry_request_on_403():
+    service = ServiceFactory.create(
+        api_root="https://example.com/",
+        auth_type=AuthTypes.zgw,
+        client_id="my-client-id",
+        secret="my-secret",
+    )
+
+    with requests_mock.Mocker() as m:
+        m.get(
+            "https://example.com/",
+            status_code=403,
+        )
+
+        with freeze_time("2024-11-27T10:00:00+02:00"):
+            initial_time = int(time.time())
+            client = build_client(service, NLXWithRefreshClient)
+
+        with freeze_time("2024-11-27T15:00:00+02:00"):  # 5h later
+            later_time = int(time.time())
+            with client:
+                client.request("GET", "https://example.com/")
```

Comment:
```suggestion
                client.get("https://example.com/")
```
unless there's a reason to use the low-level `request` method?